src/BDCS/API/V0.hs

```diff
@@ -129,3 +135,152 @@ errTest = throwError myError
   where
     myError :: ServantErr
     myError = createApiError err503 "test_api_error" "This is a test of an API Error Response"
+
+data RecipesListResponse = RecipesListResponse {
+    recipes     :: [T.Text],
+    offset      :: Int,
+    limit       :: Int,
+    total       :: Int
+} deriving (Generic, Show, Eq)
+instance ToJSON RecipesListResponse
+instance FromJSON RecipesListResponse
+
+-- | /api/v0/recipes/list
+-- List the names of the available recipes
+--
+-- >  {
+-- >      "recipes": [
+-- >          "development",
+-- >          "glusterfs",
+-- >          "http-server",
+-- >          "jboss",
+-- >          "kubernetes",
+-- >          "octave",
+-- >      ],
+-- >      "offset": 0,
+-- >      "limit": 20,
+-- >      "total": 6
+-- >  }
+recipesList :: GitLock -> T.Text -> Handler RecipesListResponse
+recipesList repoLock branch = liftIO $ RWL.withRead (gitRepoLock repoLock) $ do
+    -- TODO Figure out how to catch GitError and throw a ServantErr
+    recipes <- liftIO $ listBranchFiles (gitRepo repoLock) branch
```

Comment:
One of these liftIO calls seems like it would have to be unnecessary.

Reply:
Sure enough.